src/nvim/eval.c

```diff
@@ -21757,3 +21971,81 @@ bool eval_has_provider(char *name)
 
   return false;
 }
+
+static DictWatcher *dictwatcher_node_data(QUEUE *q)
+{
+  return QUEUE_DATA(q, DictWatcher, node);
+}
+
+static void dictwatcher_notify(dict_T *dict, const char *key,
+    typval_T *newtv, typval_T *oldtv)
+{
+  typval_T argv[2];
+  for (size_t i = 0; i < ARRAY_SIZE(argv); i++) {
+    init_tv(argv + i);
+  }
+
+  argv[0].v_type = VAR_STRING;
+  argv[0].vval.v_string = (uint8_t *)xstrdup(key);
```

Comment:
No. You are assigning a `char_u *` variable with a `uint8_t *` which is assiging a string with an integer array. I am replacing `char_u*` with `char*` when possible and hope it will disappear in the far future, but before this we have two character types: `char_u` and `char`. `uint8_t` is integer and it is not consistent to use it for a character until you want to use it with a character.